http.go

```diff
@@ -533,3 +554,28 @@ func isAllCaps(s string) bool {
 	}
 	return true
 }
+
+func readAgentsFile(path string) (*agent.Visitor, error) {
```

Comment:
The fact that this creates a visitor suggests that the function name is slightly wrong. At the very least I think it deserves a doc comment.